### ISSUE TITLE
Implement e2e tests using golang for eventing-manager

### ISSUE THREAD

**Issue by grischperl** [kyma-project/eventing-manager#74]
## Description:
Implement E2E tests using Golang for eventing-manager (follow the similar approach as in [nats-manager](https://github.com/kyma-project/nats-manager/tree/main/e2e)). 

**Tasks:**
- [x] Implement the following test cases in golang:
  - [x] Check if Eventing is configured according to the Eventing CR (Note: Eventing CR status ready -> EPP pods ready).
  - [x] Verify that cleanup works (after deleting Eventing CR -> no resources should be left behind in the cluster)
  - [x] Event publishing and receiving works. [Both backends] (Partially blocked by: https://github.com/kyma-project/eventing-manager/issues/77)
     - [x] Port forward EPP to publish events.
     - [x] use the same test cases for event types as in [fast-integration tests](https://github.com/kyma-project/kyma/blob/4431059269a285d097781c3aa2efa2fd8329f963/tests/fast-integration/eventing-test/eventing-test.js#L140):
        - [x] v1alpha1 subscriptions with CE (structured+binary) and legacy events.
        - [x] v1alpha2 subscriptions with CE (structured+binary) and legacy events.

**NOTE:** 
- Backend switching/tracing/dashboards are not part of this ticket.
- Implementing the github action pipelines to run these tests are not part of this ticket.

## Acceptance criteria:
- [x] The tests should be able to run for the specified backend NATS/EventMesh.
- [x] Developers must be able to execute these tests locally as well as create a job that runs them centrally
- [x] Clear indication of errors. Developers must not be required to scan through lengthy output to find the actual error
- [x] Have make targets to easily run the tests steps (e.g. prep, run etc.).
- [x] Add documentation on how to run tests
